### PR TITLE
provide narrative for depth-first-search graph traversal

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Some narrative explaining the purpose and meaning of the state variables computed during the depth first fold solution.

### CODE REVIEW

File: site/learn/tutorials/99problems.md [hunk 1]
```diff
@@ -2206,6 +2206,30 @@ let g = M.of_adjacency
 List.rev (M.dfs_fold g 'w' (fun acc c -> c :: acc) [])
 ```
 
+##### Depth-first search
```
Comment (Chris00) [inline site/learn/tutorials/99problems.md]: I don't think a new title is needed.

Also, this text is supposed to help the reader to understand the code.  Thus, it should be in the `>` bloc that is initially hidden.

**Comment by shayne-fletcher** [comment]
Have moved the narrative so that it leads into the code block. This seems best.

**Comment by Chris00** [comment]
I have squashed your commits.  See 7861e71.

**Comment by shayne-fletcher** [comment]
Very good. Thanks.

**Comment by shayne-fletcher** [comment]
In case there's any interest, here's a dot plot of the example graph. 
dg

I don't have the means to work on incorporating this into the narrative for myself right now (so can't issue a pull request for this.. sorry).

**Pull request closed without merging**